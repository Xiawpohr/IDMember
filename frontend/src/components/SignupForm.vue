<template>
  <v-card width="450">
    <v-container>
      <h2 class="display-2 text-xs-center pb-4">Sign Up</h2>
      <v-form
        v-model="valid"
        @submit.prevent="$emit('submitted', auth)"
      >
        <v-text-field
          v-model="email"
          :rules="[rules.email]"
          label="E-mail"
          color="teal"
          box
          required
        />
        <v-text-field
          v-model="password"
          :rules="[rules.password]"
          label="Password"
          color="teal"
          type="password"
          box
          required
        />
        <v-text-field
          v-model="confirmedPassword"
          :rules="[rules.confirmPassword]"
          label="Confirm Password"
          color="teal"
          type="password"
          box
          required
        />
        <div class="text-xs-center">
          <v-btn
            type="submit"
            color="teal"
            large
          >
            Sign Up
          </v-btn>
        </div>
      </v-form>
    </v-container>
  </v-card>
</template>

<script>
export default {
  data() {
    return {
      email: '',
      password: '',
      confirmedPassword: '',
      valid: false,
      isConfirmed: true,
      rules: {
        email: v => {
          const pattern = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
          return pattern.test(v) || 'Please enter a valid email'
        },
        password: v => {
          const pattern = /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*(_|[^\w])).+$/
          return (
            pattern.test(v) ||
            'Password must contain an upper case letter, a numeric character, and a special character'
          )
        },
        confirmPassword: v =>
          v === this.password || 'Please type the same password.'
      }
    }
  },
  computed: {
    auth() {
      return {
        email: this.email,
        password: this.password
      }
    }
  }
}
</script>
