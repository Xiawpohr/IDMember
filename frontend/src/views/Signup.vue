<template>
  <v-layout justify-center align-center style="height: 100vh">
    <v-progress-circular
      v-if="isLoading"
      :size="150"
      :width="20"
      color="teal"
      indeterminate
    />
    <div v-else>
      <SignupForm @submitted="signup" />
      <div class="mt-3 text-xs-center">
        <v-btn
          to='/login'
          flat
          color="teal"
        >
          Go to Log In
        </v-btn>
      </div>
    </div>
  </v-layout>
</template>

<script>
import SignupForm from '@/components/SignupForm.vue'

export default {
  components: {
    SignupForm
  },
  computed: {
    isLoading() {
      return this.$store.state.auth.isLoading
    },
    userId() {
      return this.$store.state.auth.id
    }
  },
  watch: {
    userId(val) {
      if (val !== null) {
        this.$router.push('/profile')
      }
    }
  },
  methods: {
    signup(auth) {
      this.$store.dispatch('auth/signup', auth)
    }
  }
}
</script>
