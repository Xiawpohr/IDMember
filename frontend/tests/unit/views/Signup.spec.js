import Vue from 'vue'
import Vuex from 'vuex'
import Vuetify from 'vuetify'
import { shallowMount } from '@vue/test-utils'
import Signup from '@/views/Signup.vue'
import SignupForm from '@/components/SignupForm.vue'

Vue.use(Vuex)
Vue.use(Vuetify)

let wrapper, state, actions, store

beforeEach(() => {
  state = {
    isLoading: false,
    id: null
  }
  actions = {
    signup: jest.fn()
  }
  store = new Vuex.Store({
    modules: {
      auth: {
        namespaced: true,
        state,
        actions
      }
    }
  })
  wrapper = shallowMount(Signup, { localVue: Vue, store })
})

it('should render the component', () => {
  expect(wrapper.html()).toMatchSnapshot()
})

it('should render SignupForm component', () => {
  expect(wrapper.contains(SignupForm)).toBe(true)
})

it('should sign up user when recieving submitted event', () => {
  const auth = {
    email: 'test@example.com',
    password: '123456'
  }
  const signupForm = wrapper.find(SignupForm)
  signupForm.vm.$emit('submitted', auth)
  expect(actions.signup).toHaveBeenCalled()
})
