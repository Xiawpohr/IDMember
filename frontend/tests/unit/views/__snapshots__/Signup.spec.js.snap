// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the component 1`] = `
<vlayout-stub tag="div" justify-center="" align-center="">
  <div>
    <signupform-stub></signupform-stub>
    <div class="mt-3 text-xs-center">
      <vbtn-stub activeclass="v-btn--active" flat="true" tag="button" type="button" to="/login" color="teal">
        Go to Log In
      </vbtn-stub>
    </div>
  </div>
</vlayout-stub>
`;
